### PR TITLE
[USM-1820] Add support for Sidekiq 7

### PR DESCRIPTION
**Pull request by leoshaw**
The configuration API changed in 6.5, so the specs need changing around configuring Sidekiq and creating a Processor.
Also the move to redis-client instead of redis means Fakeredis doesn't work as a drop-in solution for stubbing Redis, so we've added Redis to the CI for tests against versions of Ruby where Sidekiq 7 is tested.

See https://github.com/sidekiq/sidekiq/blob/main/Changes.md#650 for an overview and https://github.com/sidekiq/sidekiq/pull/5340/files#diff-27ae0d27735b8f7bd84e7e6ad97842a4d2ceb67b04259b94ba0469b802512061 for most of the changes

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -30,21 +30,34 @@
...
+    ].join(',')
+  when '7'
+    rspec_pattern = [
+      'spec/integration/sidekiq7_spec.rb',
+      'spec/loga/sidekiq7/**/*_spec.rb',
+      'spec/loga/sidekiq_spec.rb',
+    ].join(',')
   else
-    raise 'FIXME: Unknown sidekiq - update this file.'
+    raise "FIXME: Unknown sidekiq #{sidekiq_version} - update this file."
```
Comment (KirillZabrodinFC) [inline spec/spec_helper.rb:60]: Praise: Test spec
Nice, I really like this, especially the logs!

**Review by KirillZabrodinFC — COMMENTED** [review]
Looks good to me but might be worth getting a second opinion as I know very little!

File: Appraisals [hunk 2]
```diff
@@ -38,6 +38,22 @@ if Gem::Version.new(RUBY_VERSION) > Gem::Version.new('2.5.0')
   appraise 'sidekiq61' do
     gem 'sidekiq', '~> 6.1.0'
   end
+
+  appraise 'sidekiq62' do
```
Comment (bliof-fc) [inline Appraisals:42]: You should run `bundle exec appraisal generate` and commit the `gemfiles`

Overall appraisal does:

```
BUNDLE_GEMFILE=gemfiles/sidekiq7.gemfile bundle install

BUNDLE_GEMFILE=gemfiles/sidekiq7.gemfile bundle exec rspec
```

File: spec/integration/sidekiq7_spec.rb [hunk 3]
```diff
@@ -0,0 +1,193 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
```
Comment (bliof-fc) [inline spec/integration/sidekiq7_spec.rb:4]: 🤔  I think we can do

```
require 'fakeredis'

Thread.current[:sidekiq_redis_pool] = ConnectionPool.new(size: 5) { Redis.new }
```

File: spec/integration/sidekiq7_spec.rb [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+describe 'Sidekiq client logger' do
+  let(:target) { StringIO.new }
+  let(:config) { Sidekiq.instance_variable_get :@config }
```
Comment (bliof-fc) [inline spec/integration/sidekiq7_spec.rb:7]: `Sidekiq.default_configuration` should be the same https://github.com/sidekiq/sidekiq/blob/v7.0.7/lib/sidekiq.rb#L89

**Review by bliof-fc — APPROVED** [review]

**Pull request merged**